src/index.js

```diff
@@ -60,6 +64,7 @@ function App() {
 
     function sortBy(e) {
         const colToSort = e.target.value;
+        // KH: you dont have to assign it again.
```

Comment:
@KevinHong913 I read somewhere that in React we should try not to mess with the 'original copy', but just to play with a copy of it. 
Is this concept applicable here? Or does this immutability concept only referring to states?

Reply:
In class component it is true that you do not directly change state because it is consider as `this.xxx`.
In this case it's okay to do it here because it is just a normal variable until you use `setSortConfig` to change it. 
Also `sortConfig` is a object so what you're doing is really just assigning the reference to `currentSort` . So you're actually still mutating the same object data.

Reply:
So `const currentTodos = [...todos]` is not needed  in `toggleCompleted` and `handleClearItems` as well? 
Since in both cases we are 1. just using find and filter, there is never the need to edit {todos}
2. we are still actually mutating the original copy in the end